ql/src/java/org/apache/hadoop/hive/ql/ddl/table/info/show/status/formatter/ShowTableStatusFormatter.java

```diff
@@ -123,13 +123,17 @@ FileData getFileData(HiveConf conf, List<Path> locations, Path tablePath) throws
     if (!fileData.unknown) {
       for (Path location : locations) {
         try {
-          FileStatus status = fileSystem.getFileStatus(location);
+          // Use right FileSystem object for partition in customized location
+          FileSystem locationFileSystem = location.toString().startsWith(tablePath.toString()) ? fileSystem
+              : location.getFileSystem(conf);
```

Comment:
if the partition location is hdfs://ns1/tmp/hivePart and the table location is hdfs://ns1/hive/table
In this case the FileSystem is same, still you will land up doing this getFileSystem()? We could have matched the scheme & authority of the path only?

Secondly, if we want to do this only, can we use `FileUtils.isPathWithinSubtree(location, tablePath)`

Reply:
@ayushtkn Thanks for reviewing. It makes sense to match scheme and authority. Please review the updated PR.